ui/src/app/store/general/types/base.ts

```diff
@@ -224,6 +224,16 @@ export type PowerTypesState = {
   loading: boolean;
 };
 
+export type TLSCertificate =
+  | Omit<CertificateData, "private_key"> & CertificateMetadata;
+
+export type TLSCertificateState = {
+  errors: APIError;
+  data: TLSCertificate | EmptyObject | null;
```

Comment:
It seems a little unhelpful that it can return an empty object...

Reply:
Yep I've mentioned it to Anton so I might be able to remove this in the future, but this is how it is currently

Reply:
Sorry, I was not aware of what should be returned. It is changed now https://git.launchpad.net/maas/commit/?id=6c6bc1aa1f93184909823677f4251d352d804ce1

Reply:
@troyanov no worries, I should have mentioned this before you started implementing it. Anyway, thanks for addressing it so quick!